src/sql/workbench/contrib/editData/browser/editDataEditor.ts

```diff
@@ -290,12 +292,28 @@ export class EditDataEditor extends BaseEditor {
 			this._sash = this._register(new HorizontalFlexibleSash(parentElement, this._minEditorSize));
 			this._setSashDimension();
 
-			this._register(this._sash.onPositionChange(position => this._doLayout()));
+			this._register(this._sash.onPositionChange(position => this._positionCheck(position)));
 		}
 
 		this._sash.show();
 	}
 
+	/**
+	 * Checks if the current position of the sash doesn't cover the first row of the table.
+	 * If large enough, do layout. Else if too small, hide result editor to prevent crashing.
+	 */
+	private _positionCheck(position: number) {
```

Comment:
please don't add a `_` prefix to function names.  this isn't the code convention we're following in this codebase generally.  You can add `_` to private variable names if that is the convention being used in that file.  That varies from file-to-file in this codebase.